### PR TITLE
Modified sessions controller to return 401 for destroy action with no user signed in

### PR DESCRIPTION
**Pull request by aamarill**
This PR is for issue #4782  @tegon
To clarify the 3 tests added:

### 1. Authenticated non-navigational request - returns 204.
- This was the original behavior before I made any changes. I simply added this test to ensure this was not broken after my implementation.  

### 2. Unauthenticated non-navigational request - returns 401.
- This is the new behavior added.

### 3. Unauthenticated navigational request -  returns 302.
- This was the original behavior before I made any changes. I simply added this test to ensure this was not broken after my implementation.

### CODE REVIEW

**Comment by aamarill** [comment]
Looking into what went wrong with Travis CI.
I did forget to test using Mongoid and different Gemfiles, my apologies. Hoping that's all I need to account for.

File: test/controllers/sessions_controller_test.rb [hunk 1]
```diff
@@ -88,6 +88,30 @@ class SessionsControllerTest < Devise::ControllerTestCase
     assert_equal 204, @response.status
   end
 
+  test "#destroy returns 204 status if the requested format is not navigational" do
```
Comment (feliperenan) [inline test/controllers/sessions_controller_test.rb]: This behavior is already tested in the test above. IMHO changing the test name would be enough.

`#destroy doesn't set the flash and returns 204 status if the requested format is not navigational`

File: test/controllers/sessions_controller_test.rb [hunk 2]
```diff
@@ -88,6 +88,30 @@ class SessionsControllerTest < Devise::ControllerTestCase
...
+    delete :destroy, format:'json'
+    assert_equal 204, @response.status
+  end
+
+  test "#destroy returns 401 status if user is not signed in and the requested format is not navigational" do
+    delete :destroy, format: 'json'
+    assert_equal 401, @response.status
+  end
+
+  test "#destroy returns 302 status if user is not signed in and the requested format is navigational" do
```
Comment (feliperenan) [inline test/controllers/sessions_controller_test.rb]: 👏

File: app/controllers/devise/sessions_controller.rb [hunk 3]
```diff
@@ -62,7 +62,10 @@ def verify_signed_out_user
     if all_signed_out?
       set_flash_message! :notice, :already_signed_out
 
-      respond_to_on_destroy
+      respond_to do |format|
```
[the remark this answers is not in the document]
Comment (feliperenan) [reply app/controllers/devise/sessions_controller.rb]: Maybe the default `:no_content` isn't a good idea. I like the idea to have the status explicit in the caller.

```ruby
respond_to_on_destroy(status: :no_content)
respond_to_on_destroy(status: :unauthorized)
```

**Comment by tegon** [comment]
@aamarill are you still going to work on this?

I'm asking to know if I should assign it to someone else.

**Comment by aamarill** [comment]
@tegon please go ahead and assign to someone else. Thank you

**Comment by tegon** [comment]
@aamarill Since the changes were small, I committed them directly to your branch. Thanks for your contribution!

**Pull request merged**